### PR TITLE
YD-626 Prevent 500 in invalid date in URL

### PR DESCRIPTION
**Pull request by Bert-R**
Added exception handling. Along with this improved the error logging for invalid UUIDs by including the cause exception.

### CODE REVIEW

File: core/src/main/java/nu/yona/server/analysis/service/DayActivityDto.java [hunk 1]
```diff
@@ -78,7 +80,14 @@ public int getTotalMinutesBeyondGoal()
 
 	public static LocalDate parseDate(String iso8601)
 	{
-		return LocalDate.parse(iso8601, ISO8601_DAY_FORMATTER);
+		try
+		{
+			return LocalDate.parse(iso8601, ISO8601_DAY_FORMATTER);
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/analysis/service/DayActivityDto.java]: Wouldn't it be good to apply the same pattern at `WeekActivityDto.parseDate`?

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy [hunk 2]
```diff
@@ -1267,6 +1266,30 @@ class ActivityTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(richard)
 	}
 
+	def 'Try get raw activities with an invalid date'()
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: Good to have an acceptance test. We could also add a unit test for `DayActivityDto.parseDate`.

File: core/src/main/java/nu/yona/server/analysis/service/DayActivityDto.java (hunk 1 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/analysis/service/DayActivityDto.java]: Certainly! Added it.

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy (hunk 2 above)
Comment (Bert-R) [reply appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: Good point. Added it.

**Comment by Bert-R** [comment]
Pushed an update

**Pull request merged**